src/codegen/buffering_consumer.cpp

```diff
@@ -173,5 +183,15 @@ void BufferingConsumer::ConsumeResult(ConsumerContext &ctx,
   codegen.CallFunc(_BufferTupleProxy::GetFunction(codegen), args);
 }
 
+const std::vector<WrappedTuple> &BufferingConsumer::GetOutputTuples() {
+  size_t result_size = tuples_.GetSize();
+  tuples_vector_.reserve(result_size);
+  tuples_vector_.clear();
+  for (size_t i = 0; i < result_size; ++i) {
+    tuples_vector_.push_back(tuples_.Find(i));
```

Comment:
I'm concerning about the use of LockFreeArray container here since the lock-free array implementation uses only the atomic offset, it does not prevent other threads deleting items from it after you call GetSize(), therefore the Find() call can get invalid read. Better use the FindValid() call because that it checks whether the read is valid for you and it will return the invalid value after it finds no valid items left. However, it can also cause duplicated reads because the result_size you use may larger than the actual one.